diff -Naur kdebase-workspace-4.3.0.orig/plasma/shells/desktop/desktopcorona.cpp kdebase-workspace-4.3.0/plasma/shells/desktop/desktopcorona.cpp
--- kdebase-workspace-4.3.0.orig/plasma/shells/desktop/desktopcorona.cpp	2009-07-30 01:36:23.000000000 +0000
+++ kdebase-workspace-4.3.0/plasma/shells/desktop/desktopcorona.cpp	2009-08-01 00:01:32.000000000 +0000
@@ -23,6 +23,7 @@
 #include <QDir>
 #include <QGraphicsLayout>
 #include <QTimer>
+#include <QFile>
 
 #include <KDebug>
 #include <KDialog>
@@ -254,10 +255,14 @@
         applet->setGlobalShortcut(KShortcut("Alt+F1"));
     }
 
-    loadDefaultApplet("notifier", panel);
-    loadDefaultApplet("pager", panel);
-    loadDefaultApplet("tasks", panel);
+    loadDefaultApplet("stasks", panel);
     loadDefaultApplet("systemtray", panel);
+    loadDefaultApplet("digital-clock", panel);
+    loadDefaultApplet("notifier", panel);
+    
+    if (QFile::exists("/usr/share/kde4/services/plasma-applet-networkmanagement.desktop")) {
+       loadDefaultApplet("networkmanagement", panel);
+    }
 
     Plasma::DataEngineManager *engines = Plasma::DataEngineManager::self();
     Plasma::DataEngine *power = engines->loadEngine("powermanagement");
@@ -269,7 +274,6 @@
     }
     engines->unloadEngine("powermanagement");
 
-    loadDefaultApplet("digital-clock", panel);
     emit containmentAdded(panel);
 
     QTimer::singleShot(1000, this, SLOT(saveDefaultSetup()));
