Index: kdebase-runtime-4.2.0/knetattach/knetattach.cpp
===================================================================
--- kdebase-runtime-4.2.0.orig/knetattach/knetattach.cpp	2009-01-22 21:18:54.000000000 -0330
+++ kdebase-runtime-4.2.0/knetattach/knetattach.cpp	2009-01-22 21:18:54.000000000 -0330
@@ -47,7 +47,7 @@
     connect(_useEncryption, SIGNAL(toggled(bool)), this, SLOT(updatePort(bool)));
     connect(_createIcon, SIGNAL(toggled(bool)), this, SLOT(updateFinishButtonText(bool)));
     connect( this, SIGNAL(helpClicked() ), this, SLOT( slotHelpClicked() ) );
-    setWindowIcon(KIcon("knetattach"));
+    setWindowIcon(KIcon("folder-new"));
     disconnect(finishButton(), SIGNAL(clicked()), (QDialog*)this, SLOT(accept()));
     connect(finishButton(), SIGNAL(clicked()), this, SLOT(finished()));
     finishButton()->setText(i18n("Save && C&onnect"));
Index: kdebase-runtime-4.2.0/knetattach/knetattach.desktop
===================================================================
--- kdebase-runtime-4.2.0.orig/knetattach/knetattach.desktop	2009-01-22 21:18:54.000000000 -0330
+++ kdebase-runtime-4.2.0/knetattach/knetattach.desktop	2009-01-22 21:18:54.000000000 -0330
@@ -94,7 +94,7 @@
 GenericName[zh_TW]=網路資料夾精靈
 Exec=knetattach
 InitialPreference=6
-Icon=knetattach
+Icon=folder-new
 Type=Application
 Terminal=false
 X-KDE-StartupNotify=true
